### ISSUE TITLE
Secrets Manager can't find the specified secret during UpdateSecretVersionStage operation 

### ISSUE THREAD

**Issue by MarkBrook** [spulec/moto#4340]
I'm trying to test secrets manager rotation with moto server and aiobotocore. However, I'm facing issue on `UpdateSecretVersionStage` operation.

Environment:
```
Python 3.9.6
moto[server]==2.2.7
aiobotocore==1.4.1
```

I've prepared a small code sample to reproduce an error. I expect `current_version_id` to be staged as `AWSPREVIOUS` and `token` to be staged as `AWSCURRENT` (from the sample attached):
```python
import json
import uuid

import aiobotocore
import pytest


@pytest.fixture
def aiobotocore_session():
    return aiobotocore.get_session()


@pytest.fixture
def aws_kwargs():
    return {
        "region_name": "us-east-1",
        "endpoint_url": "http://moto:5000",
        "aws_access_key_id": "test",
        "aws_secret_access_key": "test",
    }


@pytest.fixture
async def secrets_manager(aiobotocore_session, aws_kwargs):
    async with aiobotocore_session.create_client("secretsmanager", **aws_kwargs) as service_client:
        yield service_client


@pytest.mark.asyncio
async def test_finish_secret_step_success(secrets_manager):
    very_secret_data = {"key": "old_value"}
    resp = await secrets_manager.create_secret(Name="Foo", SecretString=json.dumps(very_secret_data))
    arn, current_version_id = resp["ARN"], resp["VersionId"]
    
    updated_secret_data = {"key": "new_value"}
    token = str(uuid.uuid4())
    await secrets_manager.put_secret_value(SecretId=arn, ClientRequestToken=token, SecretString=json.dumps(updated_secret_data), VersionStages=['AWSPENDING'])

    await secrets_manager.get_secret_value(SecretId=arn, VersionId=current_version_id, VersionStage="AWSCURRENT")
    await secrets_manager.get_secret_value(SecretId=arn, VersionId=token, VersionStage="AWSPENDING")

    await secrets_manager.update_secret_version_stage(SecretId=arn, VersionStage="AWSCURRENT", MoveToVersionId=token, RemoveFromVersionId=current_version_id)
```

Here is an exception:
```
>       await secrets_manager.update_secret_version_stage(SecretId=arn, VersionStage="AWSCURRENT", MoveToVersionId=token, RemoveFromVersionId=current_version_id)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <aiobotocore.client.SecretsManager object at 0x10e927640>
operation_name = 'UpdateSecretVersionStage'
api_params = {'MoveToVersionId': 'd768f14b-582a-40d2-a0d7-578dc134ed9e', 'RemoveFromVersionId': '8bdb0868-d37a-4752-a3d8-9b711b8ad278', 'SecretId': 'arn:aws:secretsmanager:us-east-1:123456789012:secret:Foo-LNrXe', 'VersionStage': 'AWSCURRENT'}

    async def _make_api_call(self, operation_name, api_params):
        operation_model = self._service_model.operation_model(operation_name)
        service_name = self._service_model.service_name
        history_recorder.record('API_CALL', {
            'service': service_name,
            'operation': operation_name,
            'params': api_params,
        })
        if operation_model.deprecated:
            logger.debug('Warning: %s.%s() is deprecated',
                         service_name, operation_name)
        request_context = {
            'client_region': self.meta.region_name,
            'client_config': self.meta.config,
            'has_streaming_input': operation_model.has_streaming_input,
            'auth_type': operation_model.auth_type,
        }
        request_dict = await self._convert_to_request_dict(
            api_params, operation_model, context=request_context)
    
        service_id = self._service_model.service_id.hyphenize()
        handler, event_response = await self.meta.events.emit_until_response(
            'before-call.{service_id}.{operation_name}'.format(
                service_id=service_id,
                operation_name=operation_name),
            model=operation_model, params=request_dict,
            request_signer=self._request_signer, context=request_context)
    
        if event_response is not None:
            http, parsed_response = event_response
        else:
            http, parsed_response = await self._make_request(
                operation_model, request_dict, request_context)
    
        await self.meta.events.emit(
            'after-call.{service_id}.{operation_name}'.format(
                service_id=service_id,
                operation_name=operation_name),
            http_response=http, parsed=parsed_response,
            model=operation_model, context=request_context
        )
    
        if http.status_code >= 300:
            error_code = parsed_response.get("Error", {}).get("Code")
            error_class = self.exceptions.from_code(error_code)
>           raise error_class(parsed_response, operation_name)
E           botocore.errorfactory.ResourceNotFoundException: An error occurred (ResourceNotFoundException) when calling the UpdateSecretVersionStage operation: Secrets Manager can't find the specified secret.
```

**Comment by bblommers** [comment]
Thanks for raising this @MarkBrook - can confirm this is a bug.

Some implementation notes for whoever wants to pick this up:
Moto only stores the short name, `Foo`, but users usually pass in the full ARN.
In order to automatically handle this, Moto overrides the `__contains__` operator for the secrets-dict, and converts the ARN to `Foo`.

In the update_secrets_version_stage-method, we should check whether the ARN is in the dictionary, to ensure our overloaded method is called. We currently call the .keys()-method, so it uses the OOTB __contains__ method in the List-class

https://github.com/spulec/moto/blob/master/moto/secretsmanager/models.py#L732:L733

TLDR:
`if secret_id not in self.secrets.keys():`
should be
`if secret_id not in self.secrets():`

Looks like the same bug appears in `tag_resource` and `untag_resource`.

**Comment by bblommers** [comment]
@MarkBrook As a workaround, this should work:
```
secrets_manager.update_secret_version_stage(SecretId="Foo", ...)
```

**Comment by MarkBrook** [comment]
@bblommers thank you. Please check PR once you have time: https://github.com/spulec/moto/pull/4353
